arelle/Cntlr.py

```diff
@@ -628,7 +628,7 @@ def logRefsFileLines(refs: list[dict[str, Any]]) -> str:
     for ref in refs:
         href = ref.get("href")
         if href:
-            fileLines[href.partition("#")[0]].add(ref.get("sourceLine", 0))
+            fileLines[href.partition("#")[0]].add(ref.get("sourceLine", 0) if ref.get("sourceLine") else 0)
```

Comment:
Calling get on the dictionary twice is inefficient. Are you trying to guard against sourceLine being set to None? If so:
```suggestion
            fileLines[href.partition("#")[0]].add(ref.get("sourceLine") or 0)
```